cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/RemoteProgramRunDispatcher.java

```diff
@@ -87,11 +89,12 @@ public RemoteProgramRunDispatcher(CConfiguration cConf, MetricsCollectionService
                                     RemoteClientFactory remoteClientFactory, Store store,
                                     ProgramRunnerFactory programRunnerFactory,
                                     @RemoteExecution ProgramRunnerFactory remoteProgramRunnerFactory,
-                                    @RemoteExecution RemoteExecutionTwillRunnerService remoteTwillRunnerService) {
+                                    @RemoteExecution TwillRunnerService twillRunnerService) {
```

Comment:
Can this just be `RemoteTwillRunnerService twillRunnerService` instead of casting below?

Reply:
Currently there is a singleton binding from TwillRunnerService to RemoteTwillRunnerService in appfabric. We should be using the same singleton. We'll fix it in a followup PR. Created JIRA ticket for it: CDAP-18964